lib/Doctrine/DBAL/Platforms/SqlitePlatform.php

```diff
@@ -150,6 +150,19 @@ public function getDateDiffExpression(string $date1, string $date2) : string
         return sprintf("JULIANDAY(%s, 'start of day') - JULIANDAY(%s, 'start of day')", $date1, $date2);
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * The SQLite platform doesn't support the concept of a database, therefore, it always returns an empty string
```

Comment:
Is there no way to get the path to the DB?

Reply:
What would be the practical application of this return value? On the platforms which support databases, the current database can be changed after establishing a connection and may be explicitly used for schema introspection. That's why we need to be able to get it out of the connection. In the case of SQLite, the path is an implementation detail which may not even exist (in-memory database).

Reply:
Well, I don't think `getDatabase()` should be used at all, but if used, it is likely used for display/debug reasons.

Having this return the path (yes, even for in-memory) would allow making that clear, IMO

Reply:
I don't think you can return the path from SQL. I agree it's nice to have but as long as the underlying platform doesn't expose the path as a database, we cannot return it. Or does it?

> Well, I don't think `getDatabase()` should be used at all […]

You mean we shouldn't consider the fact that a different database can be selected after establishing a connection? Yes, that would simplify things but IIRC some platforms (e.g. `information_schema` in MySQL) require the schema/database to be explicitly specified for introspection. On the other hand, some platforms ignore the `$database` argument during introspection (e.g. `getListTableColumnsSQL()` in the SQL Server and IBM DB2 platforms).

With that said, I believe it is reasonable to keep `Connection::getDatabase()` around and use it appropriately. At the same time, the SQLite file path doesn't look like proper usage.

Reply:
Fair enough :+1: